### ISSUE TITLE
mbedtls_asn1_write_mpi() handles 0 incorrectly

### ISSUE THREAD

**Issue by hanno-becker** [Mbed-TLS/mbedtls#2166]
__Issue:__ The ASN.1 encoding of the zero integer should use a single value octet with value `0x00`. However, `mbedtls_asn1_write_mpi()` doesn't write any value octets.

__Steps to reproduce:__ Use `programs/x509/cert_write` to write a CRT with `serial=0` and inspect output via `dumpasn1`. This will show that the ASN.1 integer used for the serial number has no value octets:
```
  0 832: SEQUENCE {
  4 552:   SEQUENCE {
  8   3:     [0] {
 10   1:       INTEGER 2
       :       }
 13   0:     INTEGER
       :       Error: Object has zero length.
```

__References:__

From [A Layman's Guide to a Subset of ASN.1, BER, and DER](http://luca.ntop.org/Teaching/Appunti/asn1.html):
> BER encoding. Primitive. Contents octets give the value of the integer, base 256, in two's complement form, most significant digit first, with the minimum number of octets. The value 0 is encoded as a single 00 octet.

Relevant code from [asn1write.c](https://github.com/ARMmbed/mbedtls/blob/mbedtls-2.13.1/library/asn1write.c#L139):
```c
    // Write the MPI
    //
    len = mbedtls_mpi_size( X );

    if( *p < start || (size_t)( *p - start ) < len )
        return( MBEDTLS_ERR_ASN1_BUF_TOO_SMALL );

    (*p) -= len;
    MBEDTLS_MPI_CHK( mbedtls_mpi_write_binary( X, *p, len ) );
```
Here, [`mbedtls_mpi_size()`](https://github.com/ARMmbed/mbedtls/blob/mbedtls-2.13.1/library/bignum.c#L407) will return `0` if `X` has value `0`.

__Internal Reference:__ Treated as part of IOTSSL-2216.

**Comment by tom-cosgrove-arm** [comment]
This was fixed in #5917

**Comment by Christian-Sander** [comment]
Possibly related to this, on Mbed TLS 3.5.2 I notice that certificate generation with a "0" string parsed as mpi still generates an empty serial field:

`mbedtls_mpi_read_string( &serial, 10, "0" );`
`mbedtls_x509write_crt_set_serial( &crt, &serial );`
`//later written with mbedtls_x509write_crt_der()`

However, setting the raw value works:
`mbedtls_x509write_crt_set_serial_raw( &crt.value, (unsigned char*)"", 1);`

Also, am I supposed to let the user set the serial number or should I rather generate a random one? My application is generation of a self signed certificate for use with a local HTTPS server.
